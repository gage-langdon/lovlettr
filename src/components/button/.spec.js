import React from 'react';
import { shallow } from 'enzyme';
//component
import Button from './index';

describe('Button component', () => {
  it('Will render with text as props, but no onClick', () => {
    const wrap = shallow(<Button text={'Test 2'} />);
    expect(wrap).toMatchSnapshot();
  });

  it('Will render with onClick function', () => {
    const wrap = shallow(
      <Button text={'Test 3'} onClick={() => console.log('Test 3 success')} />
    );
    expect(wrap).toMatchSnapshot();
  });

  it('Will render with styling if primary is true', () => {
    const wrap = shallow(<Button text={'Test 4'} primary={true} />);
    expect(wrap).toMatchSnapshot();
  });
});
