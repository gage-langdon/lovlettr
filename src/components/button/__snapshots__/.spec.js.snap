// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Will render with disabled styling 1`] = `
<button
  className="button-primary "
  disabled={true}
  type="button"
>
  test
</button>
`;

exports[`Button component Will render with onClick function 1`] = `
<button
  className=" "
  disabled={false}
  onClick={[Function]}
  type="button"
>
  Test 3
</button>
`;

exports[`Button component Will render with styling if primary is true 1`] = `
<button
  className="button-primary "
  disabled={false}
  type="button"
>
  Test 4
</button>
`;

exports[`Button component Will render with text as props, but no onClick 1`] = `
<button
  className=" "
  disabled={false}
  type="button"
>
  Test 2
</button>
`;
